src/Dreamberd/Compilation/Compile.hs

```diff
@@ -59,10 +59,18 @@ compileFunction params name args body =
 compileCall :: [String] -> String -> [AST.AstNode] -> Either String [VM.Insts]
 compileCall params "=" [_, AST.Identifier iden, value] = compileAssignation params iden value False
 compileCall params "=" [AST.Identifier iden, value] = compileAssignation params iden value True
+compileCall _ "-" [AST.Number n] = Right [VM.Push $ VM.Number (-n)]
+compileCall _ "-" [AST.Float n] = Right [VM.Push $ VM.Float (-n)]
 compileCall params [op, '='] [AST.Identifier iden, value]
-    | op `elem` "+-*/" =
+    | op `elem` "+-*/%" =
```

Comment:
You could just trust the value of getBuiltinCallForOp to know if the operator you are trying to compile is a good one or not